### PR TITLE
Add jwt format checks - dot count, signature length

### PR DESCRIPTION
**Pull request by gokce-yetiser-vural**
Added format checks for JWT parsing.

### CODE REVIEW

File: jwt/jwt.c [hunk 1]
```diff
@@ -83,21 +85,34 @@ char* jwt_create(char * project_id, long long int time, NN_DIGIT * priv_key, int
...
+        return 0;
+    }  // dot_pos_1 = header_end
+    
+    char * dot_pos_2 = strchr(dot_pos_1 + 1, '.');
+    if (!dot_pos_2) {
+        mosquitto_log_printf(MOSQ_LOG_ERR, "Invalid JWT: Missing dot!");
+        return 0;
+    }  // dot_pos_2 = body_end
+
+    if (strlen(dot_pos_2 + 1) < (2 * NUMBYTES)) {
```
Comment (konstaty) [inline jwt/jwt.c:103]: Why do you compare with  2 * NUMBYTES?

Comment (gokce-yetiser-vural) [reply jwt/jwt.c:103]: Fist I was comparing with 64 but then I thought I should compare it by 2 * NUMBYTES as signature consists of signature_r and signature_s. Both worked.

Comment (konstaty) [reply jwt/jwt.c:103]: Well okay, it's good enough.  base64 encoding increases the size by around 35%, so it definitely can not be less than the raw data.

File: jwt/jwt.c [hunk 2]
```diff
@@ -83,21 +85,34 @@ char* jwt_create(char * project_id, long long int time, NN_DIGIT * priv_key, int
...
+        mosquitto_log_printf(MOSQ_LOG_ERR, "Invalid JWT: Signature too short!");
+        return 0;
+    }  // check signature length
+
+    simple_array_t jwt_header = base64_decode(jwt, dot_pos_1 - jwt + 1);
```
Comment (konstaty) [inline jwt/jwt.c]: Please double check these calculations.  The first one seems incorrect (extra +1)
Example: 
1f.
0 1 2
jwt = 0
dot_post = 2

lenght = 2 - 0 = 2

Comment (gokce-yetiser-vural) [reply jwt/jwt.c]: You are right, I was adding extra +1 to header and body decoding functions. I fixed it.

**Review by konstaty — APPROVED** [review]

**Pull request merged**